include/oneapi/mkl/blas/detail/mklgpu/onemkl_blas_mklgpu.hxx

```diff
@@ -0,0 +1,1791 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions
+* and limitations under the License.
+*
+*
+* SPDX-License-Identifier: Apache-2.0
+*******************************************************************************/
+
+// Buffer APIs
+
+ONEMKL_EXPORT void gemm(cl::sycl::queue &queue, oneapi::mkl::transpose transa,
```

Comment:
Is there difference between onemkl_blas_<backend>.hxx files where backend is mklgpu, mklcpu, netlib, and cublas?

Reply:
@mkrainiuk That is a very good point. cublas backend has differences due to copyright and missing windows support. Other three are same. I can merge these three. Does it sound ok?

Reply:
Yes, I think it's fine. Thank you!

Reply:
@mkrainiuk  I refactored the code.